src/javascript.grammar

```diff
@@ -32,8 +32,13 @@
   ExpressionStatement |
   NamedBlockCell[@dynamicPrecedence=1] { (VariableDefinition | QualifiedVariableDefinition) !cellname "=" Block } |
   NamedExpressionCell[@dynamicPrecedence=1] { (VariableDefinition | QualifiedVariableDefinition) !cellname "=" expression semi } |
-  NamedClassCell { kw<"class"> !cellname VariableDefinition (kw<"extends"> expression)? ClassBody } |
-  NamedFunctionCell { async? kw<"function"> Star? !cellname VariableDefinition ParamList Block }
+  // We have to supply rules for both named and unnamed classes and functions
```

Comment:
Got tests to pass!

![Screen Shot 2021-08-31 at 11 51 25 AM](https://user-images.githubusercontent.com/1141386/131559511-a3aeae6b-afd4-4248-9f8d-b6c1c1d220fa.png)

Turns out our `NamedClassCell` wasn't taking precedence over `ClassDeclaration`, which had `!statement` as it's first precedence marker, which is marked as `@cut` and caused it to take precedence.

To fix this, I added `!statement` to our rules too, which forces it to override `ClassDeclaration` if it's a top-level rule. However, this introduced a new issue: top level unnamed expressions were getting assigned to `NamedClassCell`/`NamedFunctionCell` with an error marker (cause of a missing name). To fix this, I added two new rules for unnamed class cells and unnamed class functions.

Reply:
makes sense, thanks for tackling this Bryan!! 🙏 

wonder if we could just get rid of the `cellname` rule at this point -- might be worth a try? not critical, just don't think it's doing anything? (could be wrong though 😅)

Reply:
Yep, totally doable and tests still pass